src/main/java/core/basesyntax/impl/StorageImpl.java

```diff
@@ -1,19 +1,50 @@
 package core.basesyntax.impl;
 
 import core.basesyntax.Storage;
+import java.util.Objects;
 
+@SuppressWarnings("unchecked")
 public class StorageImpl<K, V> implements Storage<K, V> {
+    private static final int MAX_SIZE = 10;
+    private Pair<K, V>[] pairs;
+    private int size = 0;
+
+    public StorageImpl() {
+        pairs = new Pair[MAX_SIZE];
+        size = 0;
+    }
+
     @Override
     public void put(K key, V value) {
+        int index = getIndex(key);
+        if (index != -1) {
+            pairs[index].setValue(value);
+        } else {
+            pairs[size++] = new Pair<>(key, value);
+        }
     }
 
     @Override
     public V get(K key) {
-        return null;
+        int index = getIndex(key);
+        if (index == -1) {
+            return null;
+        }
+        return pairs[index].getValue();
```

Comment:
I would recommend using a ternary operator
```suggestion
        return index == -1 ? null :  pairs[index].getValue();
```